libr/bin/format/mach0/mach0.c

```diff
@@ -3425,56 +3432,60 @@ RSkipList *MACH0_(get_relocs)(struct MACH0_(obj_t) *bin) {
 	else\
 		r_skiplist_insert (relocs, reloc);\
 } while (0)
-			case BIND_OPCODE_DO_BIND:
-				if (!threaded_binds && addr >= segmentAddress) {
-					bprintf ("Error: Malformed DO bind opcode 0x%"PFMT64x"\n", addr);
-					goto beach;
-				}
-				DO_BIND ();
-				if (!threaded_binds) {
-					addr += wordsize;
-				} else {
-					sym_ord++;
-				}
-				break;
-			case BIND_OPCODE_DO_BIND_ADD_ADDR_ULEB:
-				if (addr >= segmentAddress) {
-					bprintf ("Error: Malformed ADDR ULEB bind opcode\n");
-					goto beach;
-				}
-				DO_BIND ();
-				addr += ULEB () + wordsize;
-				break;
-			case BIND_OPCODE_DO_BIND_ADD_ADDR_IMM_SCALED:
-				if (addr >= segmentAddress) {
-					bprintf ("Error: Malformed IMM SCALED bind opcode\n");
-					goto beach;
-				}
-				DO_BIND();
-				addr += (ut64)imm * (ut64)wordsize + wordsize;
-				break;
-			case BIND_OPCODE_DO_BIND_ULEB_TIMES_SKIPPING_ULEB:
-				count = ULEB();
-				skip = ULEB();
-				for (j = 0; j < count; j++) {
-					if (addr >= segmentAddress) {
-						bprintf ("Error: Malformed ULEB TIMES bind opcode\n");
+				case BIND_OPCODE_DO_BIND:
+					if (!threaded_binds && addr >= segment_end_addr) {
```

Comment:
imho the second condition is >, not >=

Reply:
The end address is exclusive, so `>=` means it will reject addresses outside the segment. Does it make sense to allow a bind address that's pointing at the first byte after the segment?